src/components/GoldenEye/GoldenEye.tsx

```diff
@@ -1,80 +1,122 @@
-    "Using GoldenEye:\n" +
-    "Step 1: Enter a valid URL of the target.\n" +
-    "       Eg: https://www.google.com\n\n" +
-    "Step 2: Enter any additional options for the scan.\n" +
-    "       Eg: U\n\n" +
-    "Step 3: Enter any additional parameters for the scan.\n" +
-    "       Eg: W 100\n\n" +
-    "Step 4: Click Scan to commence GoldenEye's operation.\n\n" +
-    "Step 5: View the Output block below to view the results of the tool's execution.";
-
-interface FormValues {
+/**
+ * Represents the form values for the Goldeneye component.
+ */
+interface FormValuesType {
     url: string;
-    useragent: string;
+    userAgent: string;
     worker: number;
     sockets: number;
     method: string;
-    sslcheck: string;
+    sslCheck: string;
 }
 
-const DosHTTPMethod = ["get", "post", "random"];
-const SSLCheckStatus = ["Yes", "No"];
-
+/**
+ * The Goldeneye component.
+ * @returns The Goldeneye component.
+ */
 const GoldenEye = () => {
-    const [loading, setLoading] = useState(false);
-    const [output, setOutput] = useState("");
-    const [pid, setPid] = useState("");
-    const [allowSave, setAllowSave] = useState(false);
-    const [hasSaved, setHasSaved] = useState(false);
-    const [selectedmethod, setSelectedMethod] = useState("");
-    const [selectedSSLCheck, setSelectedSSLCheck] = useState("");
-
+    //Component State Variables.
+    const [loading, setLoading] = useState(false); // State variable to indicate loading state.
+    const [output, setOutput] = useState(""); // State variable to store the output of the command execution.
+    const [pid, setPid] = useState(""); // State variable to store the process ID of the command execution.
+    const [allowSave, setAllowSave] = useState(false); // State variable to indicate if saving is allowed
+    const [hasSaved, setHasSaved] = useState(false); // State variable to indicate if the output has been saved
+    const [selectedMethod, setSelectedMethod] = useState(""); //State variable to store the method selected
+    const [selectedSslCheck, setSelectedSslCheck] = useState(""); //State variable to store selected ssl check
+    const [isCommandAvailable, setIsCommandAvailable] = useState(false); // State variable to check if the command is available.
+    const [loadingModal, setLoadingModal] = useState(true); // State variable that indicates if the modal is opened.
+    const [opened, setOpened] = useState(!isCommandAvailable); // State variable to indicate loading state of the modal.
+
+    // Component Constants.
+    const title = "Goldeneye"; // Title of the component.
+    const description =
+        "GoldenEye is a HTTP DoS Test Tool. This tool can be used to test if a site is susceptible to Deny of Service (DoS) attacks. Is possible to open several parallel connections against a URL to check if the web server can be compromised"; // Description of the component.
```

Comment:
I realise Kali says Deny of Service but I reckon Denial of Service sounds better.